### PR TITLE
chore: Update RELEASE.md

### PR DESCRIPTION
**Pull request by chensun**
**Description of your changes:**
fix a code path

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by Linchin** [comment]
Found the same error around the same time, but you created the PR first! Could you please also update the link [here](https://github.com/kubeflow/pipelines/blob/e0529fe26b6339cd6751e24c7a9de48e7f6bfeef/RELEASE.md?plain=1#L202)? It's dead now.

File: RELEASE.md [hunk 1]
```diff
@@ -235,7 +235,7 @@ and then "Retry", because after waiting for previous step, artifacts are now rea
     ```bash
     git checkout $BRANCH
     git pull upstream $BRANCH
-    cd backend/api/python_http_client
+    cd backend/api/v2beta1/python_http_client
```
Comment (gkcalat) [inline RELEASE.md:238]: Should we add a comment describing how to update `v1beta1` api package?

Comment (chensun) [reply RELEASE.md:237]: Likely v1 API becomes stable at this point, we should rarely need to release v1 api package. 
In fact, if we do, we might need to do it from a old release branch.

Comment (gkcalat) [reply RELEASE.md:237]: That makes sense. Thanks!

**Review by gkcalat — APPROVED** [review]
Thanks, @chensun!

/lgtm
/approve

**Pull request merged**